pkg/discovery/metrics/metric.go

```diff
@@ -328,3 +331,28 @@ func (m EntityStateMetric) UpdateValue(existing interface{}, maxMetricPointsSize
 func GenerateEntityStateMetricUID(eType DiscoveredEntityType, id string, rType ResourceType) string {
 	return string(eType) + "-" + id + "-" + string(rType)
 }
+
+// AccumulateMultiPoints adds the metric points from the "newPortion" to the currently "accumulated".
+// Since the length of the metric point slice may be different due to pod startup and shutdown in the collection period,
+// this is not a trivial addition.  For now we will just discard any slice of metric points not covering the full
+// collection period.
+func AccumulateMultiPoints(accumulated map[ResourceType][]Point, newPortion map[ResourceType][]Point) {
```

Comment:
What do you think about collecting this data from the already processed pod entityDTOs instead.
So we currently collect the namespace metrics separately from the workers where we process the pods per node. Because of that we need to accumulate the metrics from different workers and then collate them (after all the workers finish their work) in `Newk8sNamespaceDiscoveryWorker.Do()`.
I think we would have all the needed metric values (as part of the commodities) already in the entityDTOs created prior to this, it will be simpler logic and wouldn't have the discrepency possible here (or I may be missing something).
Haiving said that, given you have already implemented all this additional logic hooked at different places, it can always be an item for the future backlog.

Reply:
I had the same thought.  I think that's a good idea; there is definitely an advantage that we don't have to convert the cpu millicore to MHz twice.  As you said I went down this path first following the namespace discovery framework, so we can revisit and refactor it in the future.

Also please note that this piece of accumulating logic is required even computing from pod entity dtos.  This logic is aggregating pod stats to the namespace.  We didn't have any implementation for that prior to this (or I'm unaware of).

Btw, thanks a lot Irfan for proactively reviewing this!  So much appreciated!

Reply:
You are welcome @chlam4. Meanwhile I agree that the functionality in this method will be needed. The overall code of collecting this data is however scattered, which is difficult in terms of maintainability. As I said this can certainly be an improvement later. Not sure how to track that improvement though, probably put it in a backlog somewhere (or simply a TODO here). WDYT?